### PR TITLE
Datamapper integration

### PR DESCRIPTION
**Pull request by gashcrumb**
Currently the component shows up, we need data in the integration though to pass along what types are in use.  Also need to add the datamapper endpoint to the config.json file.

@chirino @kcbabo FYI

### CODE REVIEW

**Comment by kahboom** [comment]
@gashcrumb Give me a holler when this is ready to be reviewed.

**Comment by gashcrumb** [comment]
Will need more styling work and some TODOs in the code, but it's a first cut of integrating this.

**Comment by gashcrumb** [comment]
Oh, once this is merged folks will want to do a `yarn install` to ensure the datamapper dependency is pulled in.

File: package.json [hunk 1]
```diff
@@ -31,6 +31,7 @@
     "angular2-toaster": "2.0.0",
     "bootstrap": "3.3.7",
     "core-js": "2.4.1",
+    "ipaas.data.mapper": "https://github.com/redhat-ipaas/datamapper-ui.git",
```
Comment (kahboom) [inline package.json]: :open_mouth:

File: src/app/app.module.ts [hunk 2]
```diff
@@ -22,6 +22,8 @@ import { ConfigService } from './config.service';
 import { UserService } from './common/user.service';
 import { log } from './logging';
 
+import { DataMapperModule } from 'ipaas.data.mapper';
```
Comment (kahboom) [inline src/app/app.module.ts]: :open_mouth:

File: src/app/integrations/edit-page/child-aware-page.ts [hunk 3]
```diff
@@ -1,9 +1,11 @@
 import { Subscription } from 'rxjs/Subscription';
 import { ActivatedRoute, Params, Router } from '@angular/router';
+import { CurrentFlow } from './current-flow.service';
 
 export abstract class ChildAwarePage {
```
Comment (kahboom) [inline src/app/integrations/edit-page/child-aware-page.ts]: How aware is this child though, really?

File: src/app/integrations/edit-page/edit-page.component.html [hunk 4]
```diff
@@ -1,16 +1,15 @@
 <ipaas-loading [loading]="!(integration | async)">
   <div class="integrations edit" *ngIf="integration | async">
-    <div class="wizard-pf-row">
+    <div [class]="getPageRow()">
```
Comment (kahboom) [inline src/app/integrations/edit-page/edit-page.component.html]: I like it..

File: src/app/integrations/edit-page/edit-page.component.ts [hunk 5]
```diff
@@ -66,9 +96,6 @@ export class IntegrationsEditPage extends ChildAwarePage implements OnInit, OnDe
   }
 
   ngOnInit() {
-    this.flowSubscription = this.currentFlow.events.subscribe((event: FlowEvent) => {
```
Comment (kahboom) [inline src/app/integrations/edit-page/edit-page.component.ts]: I'm happy with this.

File: src/app/integrations/edit-page/flow-view/flow-view-step.component.html [hunk 6]
```diff
@@ -42,7 +42,7 @@ <h4 class="modal-title pull-left">Confirm Delete</h4>
 
   <!-- Progress Menu -->
 
-  <div class="col-md-9 text" [ngSwitch]="step.stepKind">
+  <div class="col-md-9 text" *ngIf="currentStepKind !== 'datamapper'" [ngSwitch]="step.stepKind">
```
Comment (kahboom) [inline src/app/integrations/edit-page/flow-view/flow-view-step.component.html]: `if (currentStepKind) === 'datamapper' { return 'stop, drop, and roll'; }`

File: src/app/model.ts [hunk 7]
```diff
@@ -7,16 +7,16 @@ export interface BaseEntity {
 }
 
 export interface ListResult extends BaseEntity {
-    totalCount: number;
     items: Array < {} >
     ;
+    totalCount: number;
```
Comment (kahboom) [inline src/app/model.ts]: I would really love it if we could break up this huge file into multiple, smaller files.

File: src/app/model.ts [hunk 8]
```diff
@@ -106,40 +106,40 @@ export type Environments = Array < Environment > ;
...
-    integrationTemplateId: string;
-    integrationTemplate: IntegrationTemplate;
     userId: string;
     steps: Array < Step >
     ;
-    statusType: "Activated" | "Deactivated";
-    statusPhase: "Pending" | "Running" | "Succeeded" | "Failed" | "Unknown";
-    users: Array < User >
-    ;
+    statusType: "Activated" | "Deactivated" | "Draft";
```
Comment (kahboom) [inline src/app/model.ts]: Drafts FTW!

File: src/app/model.ts [hunk 9]
```diff
@@ -106,40 +106,40 @@ export type Environments = Array < Environment > ;
...
     connections: Array < Connection >
     ;
+    description: string;
+    configuration: string;
+    gitRepo: string;
+    integrationTemplate: IntegrationTemplate;
+    users: Array < User >
+    ;
+    statusPhase: "Pending" | "Running" | "Succeeded" | "Failed" | "Unknown";
```
Comment (kahboom) [inline src/app/model.ts]: YES!

File: src/app/integrations/edit-page/step-configure/data.mapper.example.host.component.ts [hunk 10]
```diff
@@ -0,0 +1,100 @@
...
+
+import { DataMapperAppComponent } from 'ipaas.data.mapper';
+
+import { CurrentFlow, FlowEvent } from '../current-flow.service';
+import { FlowPage } from '../flow-page';
+
+@Component({
+  selector: 'ipaas-data-mapper-host',
+  template: `
+    <div *ngIf="cfg.mappingInputJavaClass && cfg.mappingOutputJavaClass">
```
Comment (kahboom) [inline src/app/integrations/edit-page/step-configure/data.mapper.example.host.component.ts]: Interesting.. interesting, indeed.

**Pull request merged**